rbac/rbac/middleware.py

```diff
@@ -181,16 +218,20 @@ def process_request(self, request):  # noqa: C901
 
         """
         if is_no_auth(request):
-            request.user = User('', '')
+            request.user = IdentityHeaderMiddleware._system_user()
             return
         try:
             rh_auth_header, json_rh_auth = extract_header(request, self.header)
             username = json_rh_auth.get('identity', {}).get('user', {}).get('username')
             account = json_rh_auth.get('identity', {}).get('account_number')
             is_admin = json_rh_auth.get('identity', {}).get('user', {}).get('is_org_admin')
-        except (KeyError, JSONDecodeError) as decode_err:
+        except (KeyError, JSONDecodeError):
+            if IdentityHeaderMiddleware._system_auth(request):
+                return
+
             logger.error('Could not obtain identity on request.')
-            raise decode_err
+            HttpResponseUnauthorizedRequest()
+            return
```

Comment:
is this should be return HttpResponseUnauthorizedRequest()?

Reply:
But seems the unit test is working.

Reply:
This will be hit when there's no identity header, and no valid service PSK. Previously this returned a 500 to the client. This change just ensures it returns a 401.